samples/snippets/logs/read_job_logs.py

```diff
@@ -0,0 +1,41 @@
+#  Copyright 2022 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+# [START batch_job_logs]
+from typing import Generator, NoReturn
+
+from google.cloud import batch_v1
+from google.cloud import logging
+
+
+def get_job_logs(project_id: str, job: batch_v1.Job) -> Generator[logging.TextEntry, None, None]:
+    log_client = logging.Client(project=project_id)
+    logger = log_client.logger("batch_task_logs")
+
+    yield from logger.list_entries(filter_=f"labels.job_uid={job.uid}")
```

Comment:
Does this really need to be a seperate function? Can we not just iterate over `logger.list_entries` directly? 

If not, can we provide some context why, and maybe use an inline function so the client and log can be part of the main function?

Reply:
I put it in separate function for easier testing. I use this in test, while the printing function can be untested.

I can change it to read the stdout and see if it works that way if you prefer.

Reply:
It's an explicit goal of the samples style guide to test the output of the function itself and not use return types: https://googlecloudplatform.github.io/samples-style-guide/#result

Reply:
@kurtisvg I made the sample fit in one function and just print the logs.